components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -318,6 +318,15 @@ handler.on(PanelActions.approveHardwareTransaction.getType(), async (store: Stor
     }
 
     if (code) {
+      // TODO resolve: 'code' is an overloaded term right now. In some places it refers to the
+      // codes received from the hardware device itself.  In other places they refer to 
+      // transport errors (e.g. BridgeNotReady).  I'm adding a third case here - when
+      // the user is not authorized which is not a device code, or related to transport.
+      if (code === 'unauthorized') {
+        await store.dispatch(PanelActions.setHardwareWalletInteractionError(code))
+        return
+      }
```

Comment:
Here I check for the 'unauthorized' message sent back from the SolanaLedgerKeyring class if we need the user to grant permissions for the Panel page (we need individual permissions for both chrome:://wallet and chrome://wallet-panel).

For this panel case we dispatch an action to update the UI (instead of catching the error in the component like we do for chrome://wallet).